src/lib.rs

```diff
@@ -1877,6 +1877,13 @@ impl Connection {
                     // TODO: in the future we might want to buffer other types
                     // of undecryptable packets as well.
                     let pkt_len = b.off() + payload_len;
+
+                    // Don't add to the queue if this packet is not decryptable
+                    // because of the mismatched size.
+                    if b.len() < pkt_len {
```

Comment:
I think larger packet than `payload_len` is ok (e.g. padded at the end) but if we need to have
a strict comparision, == may be better.

Reply:
`==` would mean we break clients that send coalesced packets, or padded ones as you say, so we shouldn't do that..